tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,464 @@
+                                        floating_images=floating_tags,
+                                        manifest_results=manifest_results,
+                                        annotations=annotations)
+
+    if workers:
+        workflow.buildstep_plugins_conf = [{'name': PLUGIN_BUILD_ORCHESTRATE_KEY}]
+
+    if reactor_config_map:
+        registries_list = []
+
+        for docker_uri in registry_conf:
+            reg_ver = registry_conf[docker_uri]['version']
+            reg_secret = None
+            if 'secret' in registry_conf[docker_uri]:
+                reg_secret = registry_conf[docker_uri]['secret']
+
+            new_reg = {}
+            if reg_secret:
+                new_reg['auth'] = {'cfg_path': reg_secret}
+            else:
+                new_reg['auth'] = {'cfg_path': str(temp_dir)}
+            new_reg['url'] = 'https://' + docker_uri + '/' + reg_ver
+
+            registries_list.append(new_reg)
+
+        platform_descriptors_list = []
+        for platform in goarch:
+            new_plat = {
+                'platform': platform,
+                'architecture': goarch[platform],
+            }
+            platform_descriptors_list.append(new_plat)
+
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+        workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] =\
+            ReactorConfig({'version': 1,
+                           'registries': registries_list,
+                           'platform_descriptors': platform_descriptors_list})
+
+    runner = ExitPluginsRunner(tasker, workflow, plugins_conf)
+    results = runner.run()
+    plugin_result = results[PushFloatingTagsPlugin.key]
+
+    if expected_exception is None:
+        manifest_type, list_type = {
+            'v2': (
+                'application/vnd.docker.distribution.manifest.v2+json',
+                'application/vnd.docker.distribution.manifest.list.v2+json',
+            ),
+            'oci': (
+                'application/vnd.oci.image.manifest.v1+json',
+                'application/vnd.oci.image.index.v1+json',
+            ),
+        }[schema_version]
+
+        for registry, _ in registry_conf.items():
+            target_registry = mocked_registries[registry]
+            for image in floating_tags:
+                name, tag = image.split(':')
+
+                assert tag in target_registry.get_repo(name)['tags']
+
+                raw_manifest_list = to_text(target_registry.get_manifest(name, tag))
+                manifest_list = json.loads(raw_manifest_list, object_pairs_hook=OrderedDict)
+
+                # Check if the manifest list is sorted
+                assert json.dumps(manifest_list, indent=4, sort_keys=True,
+                                  separators=(',', ': ')) == raw_manifest_list
+                arch_list = [m['platform']['architecture'] for m in manifest_list['manifests']]
+                assert arch_list == sorted(arch_list)
```

Comment:
This is a valid test for group_manifests, but I don't think it's valid for push_floating_tags. When *creating* the manifest list, yes, we want to make sure the keys are sorted. But when pushing new tags for the existing manifest list, we have to keep it exactly as it was, however that was.

Reply:
Fixed